packages/dds/map/src/test/mocha/summary.spec.ts

```diff
@@ -0,0 +1,186 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { existsSync, readFileSync, writeFileSync } from "node:fs";
+import path from "node:path";
+import {
+	MockContainerRuntimeFactory,
+	MockFluidDataStoreRuntime,
+	MockStorage,
+} from "@fluidframework/test-runtime-utils";
+import { convertSummaryTreeToITree } from "@fluidframework/runtime-utils";
+import { DirectoryFactory, IDirectory, SharedDirectory } from "../..";
+
+const fileBase = path.join(__dirname, "../../../src/test/mocha/snapshots/");
```

Comment:
Could you look into moving packages\dds\tree\src\test\snapshots\snapshotTools.ts such that it's usable by other DDSes? I'd love to remove the duplicated file manipulation logic we have.

Specifically, I think you can use `this.currentTest.file` and do some manipulation on it to always put snapshots for a given DDS inside that DDS's `src/test/snapshots` folder by replacing the `dist` part of the path with src and truncating at the right level. You'll want some safety checks to make sure it's not putting files e.g. outside of source tree (the `dist` that you replace with `src` should be somewhere after the common ancestor of the JS in test-dds-utils, accessible via __dirname, and the current test file).

That will let us simplify a bunch of other places we do this sort of thing as well.

Reply:
I've added a sanity check for the snapshots file path. I have a concern is that the path manipulations within `packages\dds\tree\src\test\snapshots\snapshotTools.ts` assume that the snapshot file is always located in `...\test\snapshots`. However, there is an additional layer, mocha, within the map package (`...\test\mocha\snapshots`), so I leave the path manipulations for directory snapshots within the current test file. believe it would be beneficial to create a work item to refactor the hierarchy of map tests if we want to ensure the consistency of snapshot tools.